scripts/cleanup-project.sh

```
@@ -0,0 +1,142 @@
+.DS_Store
+Thumbs.db
+
+# Project specific
+.backup/
+todo_report.txt
+mock_implementations.txt
+duplicate_test_files.txt
+EOF
+# Merge with existing .gitignore
+if [ -f .gitignore ]; then
+  grep -Fxvf .gitignore .gitignore.tmp >> .gitignore
```

Shellcheck (0.10.0)

[info] Make sure not to read and write the same file in the same pipeline.

(SC2094)